src/pages/HomePage.js

```diff
@@ -1,47 +1,42 @@
-import React, { useContext } from "react";
-
+import React, { useContext, useEffect } from "react";
 import Container from "react-bootstrap/Container";
 import Col from "react-bootstrap/Col";
 import Row from "react-bootstrap/Row";
 import { Context } from "../contex";
+import getCityWeather from "../apis/weather-api";
 
 const HomePage = () => {
-  const { weather } = useContext(Context);
+  const { weather, setWeather } = useContext(Context);
+
+  useEffect(() => {
+    getCityWeather(setWeather);
```

Comment:
We should pass `city` instead of `setWeather` and we should use `setWeather` in this `useEffect`.

Reply:
const city = "London";
const cityWeather = getCityWeather(city);
setWeather(cityWeather);